### PR TITLE
TEST: Use package-wide setup and teardown to adjust numpy print options

### LINKED ISSUE #784: doctest AFNIHeader.get_data_scaling fails if ran "alone"
originally reported in https://github.com/nipy/nibabel/pull/782 but has nothing to do with the endianess, so filing separately

when I ran the full sweep using `python -m nose -s -v --with-doctest  nibabel 2>&1 | tee tests.log` that doctest was reported as passed:
```
(sid_s390x-dchroot)yoh@zelenka:~/nibabel$ grep AFNIHeader.get_data_scaling  tests.log 
Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling ... ok
``` 
but when I run only the doctest - it fails
```shell
(sid_s390x-dchroot)yoh@zelenka:~/nibabel$ python -m nose -s -v --with-doctest  nibabel/brikhead.py
Doctest: nibabel.brikhead.AFNIHeader.__init__ ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_affine ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling ... FAIL
Doctest: nibabel.brikhead.AFNIHeader.get_volume_labels ... ok
Doctest: nibabel.brikhead.AFNIImage ... ok
Doctest: nibabel.brikhead._unpack_var ... ok
Doctest: nibabel.brikhead.parse_AFNI_header ... ok

======================================================================
FAIL: Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2224, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for nibabel.brikhead.AFNIHeader.get_data_scaling
  File "/home/yoh/nibabel/nibabel/brikhead.py", line 414, in get_data_scaling

----------------------------------------------------------------------
File "/home/yoh/nibabel/nibabel/brikhead.py", line 422, in nibabel.brikhead.AFNIHeader.get_data_scaling
Failed example:
    header.get_data_scaling()
Expected:
    array([  3.88336300e-08])
Got:
    array([3.883363e-08])


----------------------------------------------------------------------
Ran 7 tests in 0.011s
```
that is numpy 1.16.2 .  Happens to me also on my laptop (regular little endian amd64) with the same numpy
```
(git)hopa:~nipy-suite/nibabel[master]git
$> python -m nose -s -v --with-doctest  nibabel/brikhead.py
Doctest: nibabel.brikhead.AFNIHeader.__init__ ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_affine ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling ... FAIL
Doctest: nibabel.brikhead.AFNIHeader.get_volume_labels ... ok
Doctest: nibabel.brikhead.AFNIImage ... ok
Doctest: nibabel.brikhead._unpack_var ... ok
Doctest: nibabel.brikhead.parse_AFNI_header ... ok

======================================================================
FAIL: Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2224, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for nibabel.brikhead.AFNIHeader.get_data_scaling
  File "/home/yoh/proj/nipy/nipy-suite/nibabel/nibabel/brikhead.py", line 414, in get_data_scaling

----------------------------------------------------------------------
File "/home/yoh/proj/nipy/nipy-suite/nibabel/nibabel/brikhead.py", line 422, in nibabel.brikhead.AFNIHeader.get_data_scaling
Failed example:
    header.get_data_scaling()
Expected:
    array([  3.88336300e-08])
Got:
    array([3.883363e-08])


----------------------------------------------------------------------
Ran 7 tests in 0.017s

FAILED (failures=1)
```

### PR DESCRIPTION
**Pull request by effigies**
Fixes #784.

### CODE REVIEW

File: nibabel/__init__.py [hunk 1]
```diff
@@ -35,16 +35,28 @@
...
+
+def setup_package():
+    """ Set numpy print style to legacy="1.13" for newer versions of numpy """
+    import nibabel as nb
+    import numpy as np
     from distutils.version import LooseVersion
-    if LooseVersion(numpy.__version__) >= LooseVersion('1.14'):
-        numpy.set_printoptions(legacy="1.13")
+    if nb._save_printopts is None:
+        nb._save_printopts = np.get_printoptions().get('legacy')
```
Comment (yarikoptic) [inline nibabel/__init__.py]: so it is not "printopts" really but a "legacyprintopt", right?

FWIW -- ATM it is just a single variable here. In DataLad we setup/teardown a few, so I just made a `_test_states` dict variable: https://github.com/datalad/datalad/blob/master/datalad/__init__.py#L92 to store all those in without breeding them at the top module level.

**Comment by yarikoptic** [comment]
<details>
<summary>FWIW doctest is passing for me locally with this change</summary> 

```shell
$> python -m nose -s -v --with-doctest nibabel/brikhead.py
Doctest: nibabel.brikhead.AFNIHeader.__init__ ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_affine ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_volume_labels ... ok
Doctest: nibabel.brikhead.AFNIImage ... ok
Doctest: nibabel.brikhead._unpack_var ... ok
Doctest: nibabel.brikhead.parse_AFNI_header ... ok

----------------------------------------------------------------------
Ran 7 tests in 0.015s

OK
```
</details>

**Comment by effigies** [comment]
> In DataLad we setup/teardown a few, so I just made a _test_states dict variable: https://github.com/datalad/datalad/blob/master/datalad/__init__.py#L92 to store all those in without breeding them at the top module level.

Nice. I went ahead and copied that approach. I think it looks cleaner overall,

**Comment by effigies** [comment]
Any further comments?

File: nibabel/__init__.py [hunk 2]
```diff
@@ -35,16 +35,28 @@
...
+    if LooseVersion(np.__version__) >= LooseVersion('1.14'):
+        if _test_states.get('legacy_printopt') is None:
+            _test_states['legacy_printopt'] = np.get_printoptions().get('legacy')
+        np.set_printoptions(legacy="1.13")
+
+def teardown_package():
+    """ Reset print options when tests finish """
+    import numpy as np
+    if _test_states.get('legacy_printopt') is not None:
+        np.set_printoptions(legacy=_test_states['legacy_printopt'])
```
Comment (yarikoptic) [inline nibabel/__init__.py]:
```suggestion
        np.set_printoptions(legacy=_test_states.pop('legacy_printopt'))
```
and remove follow up line?

File: nibabel/__init__.py [hunk 3]
```diff
@@ -35,16 +35,28 @@
...
+        _test_states['legacy_printopt'] = None
```
Comment (effigies) [inline nibabel/__init__.py]:
```suggestion
```

**Comment by effigies** [comment]
Thanks for the review.

**Pull request merged**